### PR TITLE
Cache rewrite

### PR DESCRIPTION
**Pull request by queer**
Big changes. Note that this has an effect on a lot of "subsystems" / "commands" / etc., and as such should probably be tested more than that I had it in me to do after being up for far too long x-x

### CODE REVIEW

File: lib/alice/cache/cache.ex [hunk 1]
```diff
@@ -210,6 +271,80 @@ defmodule Alice.Cache do
...
+      vals = keys |> Enum.map(fn(x) -> ":" <> x end) |> Enum.join(", ")
+      # And now the insanity begins! Let's pray we're not about to get
+      # CQL-injected!
+      stmt = "INSERT INTO #{table} (#{cols}) VALUES (#{vals})"
+      # Then we need to make our mapping object
+      obj = keys |> Enum.reduce(%{}, fn(x, acc) -> 
+                Map.put(acc, x, data[x])
+              end)
+      Logger.debug "[CACHE] [DB] Running query: #{stmt} with data #{inspect obj, pretty: true}"
+      prep = Xandra.prepare! :cache, stmt, pool: DBConnection.Poolboy
```
Comment (Shredder121) [inline lib/alice/cache/cache.ex]: `Xandra.pretend! "that we're using prepared statements"`? 🤔

Comment (queer) [reply lib/alice/cache/cache.ex]: @Shredder121 No, because it is using prepared statements. Just building them is, uh, less than sane.

Comment (queer) [reply lib/alice/cache/cache.ex]: @Shredder121 No, because it is using prepared statements. Just building them is, uh, less than sane.

**Comment by queer** [comment]
Oh right, other than switching Mongo-based caches to Cassandra, this also changes to use [nats](https://nats.io) (specifically nats-streaming) instead of redis for event queuing.

**Pull request merged**